portal/templates/teacher_views/allGrades.html

```diff
@@ -0,0 +1,44 @@
+{% extends 'layouts/base.html' %}
+{% block title %}All Grades{% endblock %}
+
+{% block content %}
+
+<header>
+  <h2>Total Grades for Students in {{ session['session_name'] }}</h2>
+
+  <a class='back' href="{{ url_for('sessions.session_manage', course_id=course['course_num'] )}}">Back</a>
+</header>
+
+<h3>Grades</h3>
+
+<div id='style_tables'>
+<table id='student'>
+
+  <tr>
+    <th>Student</th>
+  </tr>
+
+  {% for student in students %}
+  <tr>
+    <td>{{ student['name'] }}</td>
+  </tr>
+  {% endfor %}
+
+</table>
+
+<table id='grade'>
+
+  <tr>
+    <th>Grade</th>
+  </tr>
+
+  {% for value in letter_grade %}
+  <tr>
+    <td>{{ value }}</td>
+  </tr>
+  {% endfor %}
+
+</table>
```

Comment:
Why are students' names and grades in 2 different tables?  That doesn't seem very semantically correct.  They should be 2 columns of the same table

If they need to be because of the separate `for` loops, maybe try putting each student and their grade in a list of tuples and passing that into the template instead:

`[ (student, letter_grade), (student2, letter_grade), (etc.) ]`

That way you only have to loop over one variable in the template and can keep the grade book in one table